java/servers/test/org/xtreemfs/test/mrc/ReselectOSDsTest.java

```diff
@@ -0,0 +1,243 @@
+        testEnv.shutdown();
+        Logging.logMessage(Logging.LEVEL_DEBUG, this, BufferPool.getStatus());
+    }
+
+    @Test
+    public void testDistinctReselection() throws Exception {
+        String volumeName = "testDistinctReselection";
+        String fileName = "testfile";
+
+        Options options = new Options();
+        Client client = ClientFactory.createClient(dirServiceAddress, userCredentials, null, options);
+        client.start();
+
+        client.createVolume(mrcServiceAddress, auth, userCredentials, volumeName);
+        Volume volume = client.openVolume(volumeName, null, options);
+        volume.setDefaultReplicationPolicy(userCredentials, "/", ReplicaUpdatePolicies.REPL_UPDATE_PC_WQRQ, 2, 0);
+        volume.setOSDSelectionPolicy(
+                userCredentials,
+                Helper.policiesToString(new OSDSelectionPolicyType[] {
+                        OSDSelectionPolicyType.OSD_SELECTION_POLICY_FILTER_DEFAULT,
+                        OSDSelectionPolicyType.OSD_SELECTION_POLICY_SORT_UUID }));
+
+        
+        AdminFileHandle file = (AdminFileHandle) volume.openFile(userCredentials, fileName,
+                Helper.flagsToInt(SYSTEM_V_FCNTL.SYSTEM_V_FCNTL_H_O_CREAT, SYSTEM_V_FCNTL.SYSTEM_V_FCNTL_H_O_RDWR),
+                0777);
+
+        byte[] dataIn = SetupUtils.generateData(256 * 1024).getData();
+        file.write(userCredentials, dataIn, 256 * 1024, 0);
+        List<Replica> replicas1 = file.getReplicasList();
+        file.close();
+
+        volume.setOSDSelectionPolicy(
+                userCredentials,
+                Helper.policiesToString(new OSDSelectionPolicyType[] {
+                        OSDSelectionPolicyType.OSD_SELECTION_POLICY_FILTER_DEFAULT,
+                        OSDSelectionPolicyType.OSD_SELECTION_POLICY_SORT_UUID,
+                        OSDSelectionPolicyType.OSD_SELECTION_POLICY_SORT_REVERSE }));
+
+        xtreemfs_reselect_osdsRequest input = xtreemfs_reselect_osdsRequest.newBuilder()
+                .setVolumeName(volumeName).setPath(fileName).build();
+        RPCResponse<xtreemfs_reselect_osdsResponse> rpcResponse = mrcClient.xtreemfs_reselect_osds(mrcAddress, auth,
+                userCredentials, input);
+        xtreemfs_reselect_osdsResponse response = rpcResponse.get();
+        rpcResponse.freeBuffers();
+
+        // Wait for the new XLocSet to install (will probably take at least leaseTimeout seconds)
+        Thread.sleep(20 * 1000);
+
+        file = (AdminFileHandle) volume.openFile(userCredentials, fileName,
+                Helper.flagsToInt(SYSTEM_V_FCNTL.SYSTEM_V_FCNTL_H_O_RDWR));
+        List<Replica> replicas2 = file.getReplicasList();
+
+        printReplicaList(replicas1);
+        printReplicaList(replicas2);
+
+        HashSet<Replica> replicaSet = new HashSet<Replica>();
+        replicaSet.addAll(replicas1);
+        replicaSet.addAll(replicas2);
+
+        assertEquals(replicas1.size() + replicas2.size(), replicaSet.size());
```

Comment:
If the two sets replicas1 and replicas2 equal, than the HashSet will contain exactly replicas1.size() == replicas2.size() elements and the the assertion `replicas1.size() + replicas2.size(), replicaSet.size()` will fail. 
The assertion will always fail if there is a common element to replicas1 and replicas2, since 
A ∩ B ≠ ∅ ⇒ |A| + |B| > |A ∩ B|

Is it enough if i add a comment to code, or should i make it more explicit by checking f.ex
`for (Replica r: replicas2) { 
   assertFalse(replicas1.contains(r));
}`